static/code/svg.js

```diff
@@ -107,5 +117,144 @@ svgEditor.getString = function() {
+  var selected = svgEditor.canvas.getSelectedElems();
+  if (selected.length !== svgEditor.updateToolbox.oldSelectedCount_) {
+    var actions = document.getElementById('selected-actions');
+    var singleActions = document.getElementById('selected-single-actions');
+    actions.style.display = selected.length > 0 ? 'block' : 'none';
+    singleActions.style.display = selected.length === 1 ? 'block' : 'none';
+    svgEditor.updateToolbox.oldSelectedCount_ = selected.length;
+  }
+  if (selected.length === 1) {
+    var element = selected[0];
+    document.getElementById('convertpath-action').style.display =
+        element.tagName == 'path' ? 'none' : 'block';
+
+    var fillStroke = svgEditor.getFillStroke(element);
+    var fill = fillStroke[0];
+    var stroke = fillStroke[1];
+
+    var fillRect = document.getElementById('fillRect');
+    var fillNoneStyle = document.getElementById('fillNone').style;
+    if (fill === 'fillNone') {
+      fillNoneStyle.display = 'inline';
+      fillRect.setAttribute('class', 'fillWhite');
+    } else {
+      fillNoneStyle.display = 'none';
+      fillRect.setAttribute('class', fill);
+    }
+
+    var strokeRect = document.getElementById('strokeRect');
+    if (stroke === 'strokeNone') {
+      strokeRect.setAttribute('stroke-width', '1');
+      strokeRect.setAttribute('stroke', '#d40000');
+      strokeRect.setAttribute('class', '');
+    } else {
+      strokeRect.setAttribute('stroke-width', '4');
+      strokeRect.setAttribute('stroke', '');
+      strokeRect.setAttribute('class', stroke);
+    }
+  }
+};
+
+svgEditor.updateToolbox.oldMode_ = '';
+svgEditor.updateToolbox.oldSelectedCount_ = 0;
+
+/**
+ * Find the fill and stroke for an element.
+ * @param {!Element} element Element with style.
+ * @return {!Array<string>} Array with normalized fill and stroke strings.
+ */
+svgEditor.getFillStroke = function(element) {
+  if (!element) throw TypeError('Element not provided');
+  var classes = element.getAttribute('class') || '';
+  classes = classes.replace(/Gray/g, 'Grey');
+  var fill = classes.match(/\b(fill(None|White|Black|Grey))\b/);
+  fill = fill ? fill[1] : 'fillNone';
+  var stroke = classes.match(/\b(stroke(None|White|Black|Grey))\b/);
+  stroke = stroke ? stroke[1] : 'strokeBlack';
+  return [fill, stroke];
+};
+
+/**
+ * Rotate the fill in the currently selected object between allowed options.
+ */
+svgEditor.changeFill = function() {
+  var element = svgEditor.canvas.getSelectedElems()[0];
+  var fill = svgEditor.getFillStroke(element)[0];
+  var fillOptions = ['fillNone', 'fillBlack', 'fillGrey', 'fillWhite'];
+  var fillIndex = Math.max(fillOptions.indexOf(fill), 0);
+  fillIndex++;
+  fillIndex %= fillOptions.length;
+  var classes = element.getAttribute('class') || '';
+  classes = classes.replace(/\bfill\w+\b/g, '');
+  classes += ' ' + fillOptions[fillIndex];
+  classes = classes.replace(/\s+/g, ' ').trim();
```

Comment:
This seems like a terrible hack.  I guess the alternative would be to use jQuery, though…